core/utils/memory.hpp

```diff
@@ -306,12 +306,13 @@ struct OnHeapTracked final : Base {
     : Base{std::forward<Args>(args)...}, rm_{rm} {}
 
  private:
-  IResourceManager& rm_;
   void Destroy() const noexcept final {
     auto& rm = rm_;
     delete this;
     rm.Decrease(sizeof(*this));
   }
+
+  IResourceManager& rm_;
 };
```

Comment:
Same codestyle as in whole file and most others
private section firstly contains methods and then after at least one empty line members